### PR TITLE
Fixed deadlock when executing fetch request

### PR DESCRIPTION
**Pull request by nextgeniuspro**
Fixed deadlock when executing fetch request from main and background threads simultaneously. As Apple docs say about using **performBlockAndWait**:
_"The one exception is: if your code is executing on the main thread, you can invoke methods on the main queue style contexts directly instead of using the block based API."_

### CODE REVIEW

**Comment by yevgeniy-logachev** [comment]
Strange that CI finished building with failure status. These changes cannot affect on compilation result

**Comment by ssuhanov** [comment]
Image

File: MagicalRecord/Categories/NSManagedObject/NSManagedObject+MagicalRecord.m [hunk 1]
```diff
@@ -47,19 +47,29 @@ + (NSUInteger) MR_defaultBatchSize
...
-	return results;	
+    };
+    
+    if (context.concurrencyType == NSMainQueueConcurrencyType && [NSThread isMainThread])
+    {
+        block();
+    }
+    else
+    {
+        [context performBlockAndWait:^{
```
Comment (akkrat) [inline MagicalRecord/Categories/NSManagedObject/NSManagedObject+MagicalRecord.m]: [context performBlockAndWait:block];

**Comment by Coeur** [comment]
@yevgeniy-logachev how to reproduce the deadlock ? can you demonstrate it with a piece of code?

**Comment by Coeur** [comment]
@yevgeniy-logachev could you at least rebase your branch to the current tip to see if the Travis builds can pass?

**Pull request closed without merging**